core/qthelper.cpp

```diff
@@ -43,12 +43,8 @@ QString weight_string(int weight_in_grams)
 {
 	QString str;
 	if (get_units()->weight == units::KG) {
-		int gr = weight_in_grams % 1000;
-		int kg = weight_in_grams / 1000;
-		if (kg >= 20.0)
-			str = QString("%1").arg(kg + (gr >= 500 ? 1 : 0));
-		else
-			str = QString("%1.%2").arg(kg).arg((unsigned)(gr + 50) / 100);
+		double kg = (double) weight_in_grams / 1000.0;
+		str = QString("%1").arg(kg, 0, 'f', kg >= 20.0 ? 0 : 1);
```

Comment:
but that doesn't round at all... and will print 1.99kg as 1.9kg when it really should be 2.0kg
Yes, it's MORE correct than the bogus logic we have right now (so I'll take it), but I'd love a follow up that correctly rounds (and yes, that's a bit tricky with our odd 20kg threshold - I'm sure you can do this :-) )